Алёшин_173-2_lc1a

pntn13 opened the pull request.

pntn13 committed:
add task again

ifsmirnov commented on main.cpp:
```diff
@@ -0,0 +1,116 @@
+
+    ~List() {
+        if (next != nullptr) {
+            delete next;
```
К слову, delete nullptr -- валидная операция.

ifsmirnov commented on main.cpp:
```diff
@@ -0,0 +1,116 @@
+}
+
+ostream& operator<<(ostream& out, List* a) {
+    for (auto i = a; i != nullptr; i = i->next) {
```
Дополнительная переменная тут не нужна.

ifsmirnov commented on main.cpp:
```diff
@@ -0,0 +1,116 @@
+    cin >> n;
+    int x;
+    cin >> x;
+    List* a = new List(x);
```
Не надо заставлять работать пользователя с указателями. Стоит сделать
1) возможность работать с пустым списком
2) метод/функцию push_back.
Пока что пользователю, которого заставляют вручную прописывать fin->next, будет грустно.

ifsmirnov commented on main.cpp:
```diff
@@ -0,0 +1,116 @@
+    if (a->x > b->x) {
+        swap(a, b);
+    }
+    List* res = new List(a->x);
```
Обработка первого элемента как-то отдельно выглядит большим костылём.

ifsmirnov replied:
В функциях merge и sort не надо вызывать new. Они не должны выделять память на куче, нужно только перекидывать указатели на next.

pntn13 committed:
update task1

исправил то чо надо было исправить --- на самом деле переписал почти всё заново
